### ISSUE TITLE
Handle overlapping vertices in `PolySlab`

### ISSUE THREAD

**Issue by tylerflex** [flexcompute/tidy3d#1512]
It seems that if two vertices in a PolySlab are overlapping, then we don't error. Is this expected behavior? For adjoint it will cause some errors. So I am wondering whether to
a) validate it in PolySlab
b) validate it in JaxPolyslab
c) allow for both cases, but then handle adjoint gradients differently with this edge case.

Any thoughts @weiliangjin2021 ?

**Comment by weiliangjin2021** [comment]
> Do we only do that in C++? I think I recall we also had such a function in python. If so then maybe the adjoint just needs to work with the "fixed" polygons instead of the raw one.

We do in both frontend and backend, as the backend deals with raw vertices data.

**Comment by weiliangjin2021** [comment]
Here is the function in the frontend: https://github.com/flexcompute/tidy3d/blob/118e3d1c95b4c5aab10f36d55b68eb733936fb70/tidy3d/components/geometry/polyslab.py#L1065-L1078

**Comment by tylerflex** [comment]
Hm, so adjoint just uses the raw `.vertices` currently. So I guess this correction is probably just never getting applied.

**Comment by momchil-flex** [comment]
Right, so just using those `_proper_vertices` should fix the error and hopefully work right off the bat.

**Comment by tylerflex** [comment]
well, what worries me is that this needs to be converted to use `jax.numpy.ndarray`

**Comment by momchil-flex** [comment]
You can use it freely inside a custom vjp right? Which is where the error happens.

**Comment by tylerflex** [comment]
it's a bit complicated. so let's say we use `proper_vertices` inside of the VJP after casting to `np`, how do we then map these vertices to the original `jax` ones to store gradients? I think it will require a bit more thought on my part to implement properly. I might just say that for overlapping vertices (ie if the length between vertices is 0) then the gradient is just 0 with respect to that edge.

**Comment by momchil-flex** [comment]
Ohh right yeah I see. Makes sense what you suggest.

**Comment by tylerflex** [comment]
Made a PR to fix this. #1513
